test/tests/functional/pbs_test_tpp.py

```diff
@@ -0,0 +1,128 @@
+# coding: utf-8
+
+# Copyright (C) 2003-2020 Altair Engineering, Inc. All rights reserved.
+# Copyright notice does not imply publication.
+#
+# ALTAIR ENGINEERING INC. Proprietary and Confidential. Contains Trade Secret
+# Information. Not for use or disclosure outside of Licensee's organization.
+# The software and information contained herein may only be used internally and
+# is provided on a non-exclusive, non-transferable basis. License may not
+# sublicense, sell, lend, assign, rent, distribute, publicly display or
+# publicly perform the software or other information provided herein,
+# nor is Licensee permitted to decompile, reverse engineer, or
+# disassemble the software. Usage of the software and other information
+# provided by Altair(or its resellers) is only as explicitly stated in the
+# applicable end user license agreement between Altair and Licensee.
+# In the absence of such agreement, the Altair standard end user
+# license agreement terms shall govern.
+
+from tests.functional import *
+import socket
+
+
+@tags('comm')
+class TestTPP(TestFunctional):
+    """
+    Test suite consists of tests to check the functionality of pbs_comm daemon
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        self.exec_path = os.path.join(self.pbs_conf['PBS_EXEC'], "bin")
+
+    def submit_job(self, set_attrib=None, exp_attrib=None, sleep=10,
+                   job_script=False, interactive=False):
+        j = Job(TEST_USER)
+        if set_attrib:
+            j.set_attributes(set_attrib)
+        if interactive:
+            j.interactive_script = [('hostname', '.*'),
+                                    ('export PATH=$PATH:%s' %
+                                     self.exec_path, '.*'),
+                                    ('qstat', '.*')]
+        j.set_sleep_time(sleep)
+        if job_script:
+            pbsdsh_path = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                       "bin", "pbsdsh")
+            script = "#!/bin/sh\n%s sleep 10" % pbsdsh_path
```

Comment:
if job_script is true, sleep value passed is not used. I do not see a reason for them to be mutually exclusive.